bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -362,57 +353,6 @@ impl Script {
     pub fn dust_value(&self) -> crate::Amount { self.minimal_non_dust() }
 
-    /// Returns the minimum value an output with this script should have in order to be
-    /// broadcastable on today's Bitcoin network.
-    ///
-    /// Dust depends on the -dustrelayfee value of the Bitcoin Core node you are broadcasting to.
-    /// This function uses the default value of 0.00003 BTC/kB (3 sat/vByte).
-    ///
-    /// To use a custom value, use [`minimal_non_dust_custom`].
-    ///
-    /// [`minimal_non_dust_custom`]: Script::minimal_non_dust_custom
-    pub fn minimal_non_dust(&self) -> crate::Amount {
-        self.minimal_non_dust_inner(DUST_RELAY_TX_FEE.into())
-    }
-
-    /// Returns the minimum value an output with this script should have in order to be
-    /// broadcastable on today's Bitcoin network.
-    ///
-    /// Dust depends on the -dustrelayfee value of the Bitcoin Core node you are broadcasting to.
-    /// This function lets you set the fee rate used in dust calculation.
-    ///
-    /// The current default value in Bitcoin Core (as of v26) is 3 sat/vByte.
-    ///
-    /// To use the default Bitcoin Core value, use [`minimal_non_dust`].
-    ///
-    /// [`minimal_non_dust`]: Script::minimal_non_dust
-    pub fn minimal_non_dust_custom(&self, dust_relay_fee: FeeRate) -> crate::Amount {
-        self.minimal_non_dust_inner(dust_relay_fee.to_sat_per_kwu() * 4)
-    }
-
-    fn minimal_non_dust_inner(&self, dust_relay_fee: u64) -> crate::Amount {
-        // This must never be lower than Bitcoin Core's GetDustThreshold() (as of v0.21) as it may
-        // otherwise allow users to create transactions which likely can never be broadcast/confirmed.
-        let sats = dust_relay_fee
-            .checked_mul(if self.is_op_return() {
-                0
-            } else if self.is_witness_program() {
-                32 + 4 + 1 + (107 / 4) + 4 + // The spend cost copied from Core
-                    8 + // The serialized size of the TxOut's amount field
-                    self.consensus_encode(&mut sink()).expect("sinks don't error") as u64 // The serialized size of this script_pubkey
-            } else {
-                32 + 4 + 1 + 107 + 4 + // The spend cost copied from Core
-                    8 + // The serialized size of the TxOut's amount field
-                    self.consensus_encode(&mut sink()).expect("sinks don't error") as u64 // The serialized size of this script_pubkey
-            })
-            .expect("dust_relay_fee or script length should not be absurdly large")
-            / 1000; // divide by 1000 like in Core to get value as it cancels out DEFAULT_MIN_RELAY_TX_FEE
-                    // Note: We ensure the division happens at the end, since Core performs the division at the end.
-                    //       This will make sure none of the implicit floor operations mess with the value.
-
-        crate::Amount::from_sat(sats)
-    }
-
```

Comment:
Why only these methods and not the others?

I was thinking that at first iteration we should only have `from_bytes` and `as_bytes` inherent methods on `Script` and put everything else into a single extension trait. (Maybe also except conversions between `Script` and `ScriptBuf`.)

Reply:
Ok, I think I finally understand your position. To check I do, you want to move to `primitives` just the data types and the bare necessities (eg constructors, getters, setters) to use them. Then very carefully only add other functions we are 100% sure are stable, and that is how you have confidence that we can not break `primitives` for a very long time. Have I got it?

Reply:
Yes.

Reply:
I'm happy to do this so that we can move forward, but FWIW I'd like to start moving other methods back out of extension traits into inherents ASAP. But I guess we can do it piecemeal starting with the "most obviously stable" ones, and it'd be a good exercise to triage our API as we move stuff.